### DISCUSSION TITLE
Самостоятельное редактирование расписания

### DISCUSSION THREAD

**Discussion by 0niel** [0niel/university-app#223] [Идеи]
**Проблема**
Парсер не всегда способен отработать точно, бывают редкие случаи, когда расписание для одного или группы предметов отображается неправильное. Такие случаи замечались и в приложении Mirea Assistant.

**Решение**
Нужно сделать возможность редактирования расписания предмета. При нажатии на кнопку "редактировать расписание предмета", у человека будет открыто диалоговое окно с возможностью выбора множества недель, на которых этот предмет проходит, а также отредактировать пару, на которой этот предмет. После редактирования у человека с помощью ScheduleLocalDatasource будет сохраняться локально отредактированная версия расписания с пометкой о том, что оно редактировалось. При обновлении с удалённого репозитория необходимо спрашивать разрешение на перезапись локального репозитория, чтобы человек не потерял свои изменения.

**Альтернатива**
Возможно хорошей альтернативой локальному сохранению будет отправка расписания на сервер и премодерацию расписания. Администратор просмотрит изменения пользователя и выяснит, является ли то расписание, которое отдаётся сервером, ошибочным. Если является, то администратор предпримет действия, для решения проблемы (фикс парсера). На время фикса у людей будет подгружаться то расписание, которое было отправлено клиентом (изменённое).

**Comment by ValeryVerkhoturov** [comment]
Сделать локальный форк существующего расписания, изменить, опубликовать (отправить на сервер), распространить по ссылке

**Reply by 0niel** [reply]
Классная идея!
